assets/app/scripts/services/data.js

```diff
@@ -270,22 +276,29 @@ angular.module('openshiftConsole')
         return;
       }
 
-      self.create(resource, null, object, context, opts).then(
-        function (data) {
-          // include the original object, so the error handler can display the kind/name
-          data.object = object;
-          successResults.push(data);
-          remaining--;
-          _checkDone();
-        },
-        function (data) {
-          // include the original object, so the handler can display the kind/name
-          data.object = object;
-          failureResults.push(data);
-          remaining--;
-          _checkDone();
-        }
-      );
+      var success = function(data) {
+        // include the original object, so the error handler can display the kind/name
+        data.object = object;
+        successResults.push(data);
+        remaining--;
+        _checkDone();
+      };
+      var failure = function(data) {
+        // include the original object, so the handler can display the kind/name
+        data.object = object;
+        failureResults.push(data);
+        remaining--;
+        _checkDone();
+      };
+
+      switch(action) {
+      case "update":
+        self.update(resource, object.metadata.name, object, context, opts).then(success, failure);
+        break;
+      default:
```

Comment:
I know I had this in my code, but we should probably have an explicit case for "create" and have a default case that results in an error. It would protect us against typos, etc.